docs/wallet_browser_extension/provider/api_documentation.md

```diff
@@ -0,0 +1,448 @@
+
+```ts
+interface GetActiveAccountRequest {
+  method: "get_active_account";
+}
+type GetActiveAccountResult = string | undefined;
+```
+
+### Example of usage
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "get_active_account";
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.getActiveAccount();
+    ```
+
+## Get Active Account Balance
+
+Returns the balance of the active account in the smallest unit (e.g. [Gray][5]) of the token.
+
+!!! warning
+    The `get_active_account_balance` should have permission granted; otherwise,
+    the `User Rejected Request 4001` error will be thrown.
+    Please check [Permission][4] for more detail.
+
+```ts
+interface GetActiveAccountBalanceRequest {
+  method: "get_active_account_balance";
+}
+type GetActiveAccountBalanceResult = string;
+```
+
+### Example of usage
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "get_active_account_balance";
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.getActiveAccountBalance();
+    ```
+
+## Get Active Acconut Latest Transactions
+
+Returns the latest transactions of the active account.
+
+### Params
+
+- `limit`: The number of transactions to return. The default value is `1`.
+- `order`: The order of transactions to return. The default value is `desc`.
+
+!!! warning
+    The `get_active_account_latest_transactions` should have permission granted; otherwise,
+    the `User Rejected Request 4001` error will be thrown.
+    Please check [Permission][4] for more detail.
+
+```ts
+interface GetActiveAcconutLatestTransactionsRequest {
+  method: "get_active_account_latest_transactions";
+  params?: {
+    limit?: number;
+    order?: "asc" | "desc";
+  };
+}
+
+interface TransactionSummary {
+  amount: string;
+  block_hash: string;
+  commands: string[];
+  command: string;
+  exit_status?: string;
+  gas_consumed: string;
+  gas_limit: string;
+  hash: string;
+  nonce: string;
+  number: string;
+  signer: string;
+  status: number;
+  timestamp: number;
+  tx_gas_consumption: string;
+}
+type GetActiveAcconutLatestTransactionsResult = TransactionSummary[];
+```
+
+### Example of usage
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "get_active_account_latest_transactions",
+      params: {
+        limit: 1,
+        order: "desc",
+      },
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.getActiveAcconutLatestTransactions({
+      limit: 1,
+      order: "desc",
+    });
+    ```
+
+## Get Current Network
+
+Returns the current network.
+
+```ts
+interface GetCurrentNetworkRequest {
+  method: "get_current_network";
+}
+type GetCurrentNetworkResult = string | undefined;
+```
+
+### Example of usage
+
+=== "RPC"
+    ```js
+    window.xpll.request({
+      method: "get_current_network";
+    });
+    ```
+
+=== "Helper Function"
+    ```js
+    window.xpll.getCurrentNetwork();
+    ```
+
+## Request Permission
```

Comment:
Its better to put a link to this section in the Permission section so its easier for users to find. 
Also, consider putting Permission section in front of API documentation section, and its API before those API that require Permission. The order of the sections are a bit jumpy right now, I have to jump to different section to find the stuff I need.

Reply:
solved by bb00eef

Reply:
solved by bb00eef

Reply:
I think API Documentation can be put in the very last, so people will read about Error, Event and Permission first and get the background info, then they start looking at APIs.

Reply:
solved by 59a14af